src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor({ health = 100, name }) {
+    this.health = health;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(bittedAnimal) {
+    if (bittedAnimal instanceof Herbivore && !bittedAnimal.hidden) {
+      bittedAnimal.health -= 50;
+    }
```

Comment:
The parameter name `bittedAnimal` may be a typo. Consider renaming it to `bittenAnimal` to better reflect the action being performed.